server/index.ts

```diff
@@ -1,13 +1,132 @@
 import express from 'express';
 import path from 'path';
+import fetch from 'cross-fetch';
+import cookieParser from 'cookie-parser';
+import dotenv from 'dotenv';
+dotenv.config();
 
-const port = 3001;
+const port = process.env.PORT || 3001;
 const app = express();
+app.use(cookieParser());
+
+if (!process.env.DISCOGS_CONSUMER_KEY) {
+  throw new Error('No Discogs Consumer Key available');
+}
+if (!process.env.DISCOGS_CONSUMER_SECRET) {
+  throw new Error('No Discogs Consumer Secret available');
+}
+
+const consumerKey = process.env.DISCOGS_CONSUMER_KEY;
+const consumerSecret = process.env.DISCOGS_CONSUMER_SECRET;
 
 app.get('/api/hello', (_request, response) => {
   response.json({ message: 'Hello from server' });
 });
 
+let oAuthRequestTokenSecret: string | null;
+let oAuthRequestToken: string | null;
+let oAuthAccessTokenSecret: string | null;
+let oAuthAccessToken: string | null;
```

Comment:
It is not possible to login with multiple clients at the same time, because they share these variables.
But it's not critical right now, but in long term, you should think about a different solution to save these tokens per user.

Reply:
For my understanding, those tokens will be overwritten when a new user logs in. It will only be critical, when two users log in at the same time. Is this correct?

Reply:
Because the access token is then stored in the cookie.